core/metrics/multiexporter.go

```diff
@@ -0,0 +1,57 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+)
+
+// MultiExporter is an interface that allows exporting spans to multiple OTLP trace exporters.
+type MultiExporter interface {
+	tracesdk.SpanExporter
+	AddExporter(exporter *otlptrace.Exporter)
```

Comment:
should be using `SpanExporter` instead since it allows any exporter to be used rather than just otlptrace epxorter. 

For example we can use [`tracetest.NewInMemoryExporter`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace/tracetest#NewInMemoryExporter) for testing (we already do this [here](https://github.com/synapsecns/sanguine/blob/c9bb0ff7e1f163f2e648b2c4367bad2f79843c27/core/metrics/tester.go#L30))